pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2704,6 +2722,58 @@ private CompletableFuture<Void> doTransactionAcknowledgeForResponse(MessageId me
         }
     }
 
+    private CompletableFuture<Void> doTransactionAcknowledgeForResponse(List<MessageId> messageIds, AckType ackType,
+                                                                        ValidationError validationError,
+                                                                        Map<String, Long> properties, TxnID txnID) {
+        BitSetRecyclable bitSetRecyclable = null;
+        long ledgerId;
+        long entryId;
+        ByteBuf cmd;
+        long requestId = client.newRequestId();
+        List<MessageIdData> messageIdDataList = new LinkedList<>();
+        for (MessageId messageId : messageIds) {
```

Comment:
Do we need to check if existing ack position conflication? It seems that the broker also will check conflication, maybe we could add conflication check at client.

Reply:
Yeah, the broker will check confliction.